### PR TITLE
Finish Admin CRUD User

### PR DESCRIPTION
**Pull request by hitshootnum1**

### CODE REVIEW

**Comment by QuangVinh93** [comment]
ok

**Comment by HUSTToanLH** [comment]
ok

File: app/controllers/admin/base_controller.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Admin::BaseController < ApplicationController
```
Comment (dieunb) [inline app/controllers/admin/base_controller.rb]: Không cần `BaseController`

File: app/controllers/admin/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    else
+      flash.now[:danger] = t "user.delete_error"
+      render :index
+    end
+  end
+
+  private
+  def ajax_respond
+    respond_to do |format|
+      format.html do
```
Comment (dieunb) [inline app/controllers/admin/users_controller.rb]: `format.html {render :index}`

File: app/controllers/admin/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  before_action :load_user, only: :index
+  before_action :load_roles, only: [:index, :new, :edit]
+  load_and_authorize_resource
+
+  def index
+    @from = params[:from]
+  end
+
+  def new
+    @from = "new"
```
Comment (dieunb) [inline app/controllers/admin/users_controller.rb]: `@action = new`

File: app/controllers/admin/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    end
+  end
+
+  def user_params
+    params.require(:user).permit :name, :email, :password,
+      :password_confirmation, :role
+  end
+
+  def load_all_users
+    @users = User.all.page params[:page]
```
Comment (dieunb) [inline app/controllers/admin/users_controller.rb]: `@users = User.page params[:page]`

**Pull request merged**